lib/cog/command/pipeline/initializer.ex

```diff
@@ -30,11 +34,22 @@ defmodule Cog.Command.Pipeline.Initializer do
 
   def handle_info({:publish, "/bot/commands", message}, state) do
     payload = Poison.decode!(message)
-    case check_history(payload, state) do
-      {true, payload, state} ->
-        {:ok, _} = ExecutorSup.run(payload)
-        {:noreply, state}
-      {false, _, state} ->
+    # Only self register when the feature is enabled via config
+    # and the incoming request is from a real user. We use
+    # the HTTP adapter as a proxy for the second condition as
+    # the only requests currently coming from that adapter are
+    # triggers.
+    self_register_flag = Application.get_env(:cog, :self_registration, false) and payload["adapter"] != "http"
```

Comment:
1. Good call on `Cog.Adapter.chat_adapter?`
1. It does but you never know how the config might get mangled. Adding `false` seemed like cheap conservative insurance.